wfe-core/src/main/java/ru/runa/wfe/chat/socket/ChatSessionHandler.java

```diff
@@ -35,37 +42,29 @@ public ChatSessionHandler(@Qualifier("sessionMessageSender") MessageSender messa
     public void addSession(Session session) {
         User user = ChatSessionUtils.getUser(session);
         Long userId = user.getActor().getId();
-        Session replacedSession = sessions.replace(userId, session);
-        if (replacedSession != null) {
-            try {
-                CloseReason closeReason = new CloseReason(CloseReason.CloseCodes.CLOSED_ABNORMALLY,
-                        "Replace " + user.getName() + "'s session");
-                replacedSession.close(closeReason);
-            } catch (IOException e) {
-                log.error("An error occurred while closing " + user.getName() + "'s session");
-            }
-            log.warn("Replace " + user.getName() + "'s session");
-        } else {
-            sessions.put(userId, session);
-        }
+
+        Set<SessionInfo> sessionSet = sessions.computeIfAbsent(userId, CREATE_SET);
+        SessionInfo sessionInfo = new SessionInfo(session);
+        sessionSet.add(sessionInfo);
     }
 
     public void removeSession(Session session) {
         Long userId = ChatSessionUtils.getUser(session).getActor().getId();
-        sessions.remove(userId, session);
+        sessions.get(userId).remove(new SessionInfo(session));
     }
 
     public void sendToSession(Session session, String message) throws IOException {
         session.getBasicRemote().sendText(message);
     }
 
-    public void sendMessage(MessageBroadcast dto) throws IOException {
+    public void sendMessage(MessageBroadcast dto) {
         sendMessage(Collections.emptySet(), dto);
     }
 
-    public void sendMessage(Collection<Long> recipientIds, MessageBroadcast dto) throws IOException {
+    public void sendMessage(Collection<Long> recipientIds, MessageBroadcast dto) {
         for (Long id : recipientIds) {
-            messageSender.handleMessage(dto, Optional.ofNullable(sessions.get(id)));
+            Set<SessionInfo> sessionsSet = sessions.get(id);
+            messageSender.handleMessage(dto, sessionsSet);
```

Comment:
```suggestion
            messageSender.handleMessage(dto, sessions.get(id));
```
Тут нет особого смысла рязделять